code/datums/craft/gun_parts.dm

```diff
@@ -190,222 +281,398 @@
 				to_chat(user, SPAN_DANGER("The serial is scribbled away."))
 
 //Grips
-/obj/item/part/gun/grip
+/obj/item/part/gun/modular/grip
 	name = "generic grip"
 	desc = "A generic firearm grip, unattached from a firearm."
+	part_overlay = "grip"
 	icon_state = "grip_wood"
 	generic = FALSE
-	bad_type = /obj/item/part/gun/grip
+	bad_type = /obj/item/part/gun/modular/grip
 	matter = list(MATERIAL_PLASTIC = 6)
 	price_tag = 100
 	rarity_value = 5
+	needs_grip_type = TRUE
+	var/type_of_grip = "wood" // Placeholder
+	part_itemstring = TRUE
+
+/obj/item/part/gun/modular/grip/New()
+	..()
+	I.weapon_upgrades[GUN_UPGRADE_DEFINE_GRIP] = type_of_grip
+	I.weapon_upgrades[GUN_UPGRADE_OFFSET] = -15 // Without a grip the gun shoots funny, players are legally allowed to not use a grip
```

Comment:
```suggestion
	I.weapon_upgrades = list(
		GUN_UPGRADE_DEFINE_GRIP = type_of_grip,
		GUN_UPGRADE_OFFSET = -15 // Without a grip the gun shoots funny, players are legally allowed to not use a grip
		)
```
The weapon upgrade stuff in the rest of the file should look something like this.

Reply:
That redefines the list and clears any other entries that previous new() iterations generate.

Reply:
Ah, I see. It's fine as is, then.